src/coffee/extend/ForceDirectedGraph.coffee

```diff
@@ -0,0 +1,154 @@
+Function::property = (prop, desc) ->
+  Object.defineProperty @prototype, prop, desc
+
+
+# Hacky Edge implementation
+class Edge
+  constructor: (source, target) ->
+    @source = source
+    @target = target
+
+  @property 'p1',
+    get: ->
+      return {
+        x: @target.x,
+        y: @target.y,
+        z: @target.z
+      }
+
+  @property 'x',
+    get: ->
+      return @source.x
+
+  @property 'y',
+    get: ->
+      return @source.y
+
+  @property 'z',
+    get: ->
+      return @source.z
+
+  clone: () ->
+    return new Edge(@source.clone(), @target.clone())
+
+
+class Vertex extends Vector
```

Comment:
Can this be a `Pair`, or do you prefer the more semantic name `displacement` instead of `p1`?